pilot/pkg/trustbundle/trustbundle.go

```diff
@@ -212,11 +212,14 @@ func (tb *TrustBundle) AddMeshConfigUpdate(cfg *meshconfig.MeshConfig) error {
 		certs := []string{}
 		endpoints := []string{}
 		for _, pemCert := range cfg.GetCaCertificates() {
-			cert := pemCert.GetPem()
-			if cert != "" {
-				certs = append(certs, cert)
-			} else if pemCert.GetSpiffeBundleUrl() != "" {
-				endpoints = append(endpoints, pemCert.GetSpiffeBundleUrl())
+			// if there is cert signer or trust domains defined for the cert, we should not use it as global trust anchor
+			if pemCert.CertSigners == nil && pemCert.TrustDomains == nil {
```

Comment:
Will this logic change sometime in the future? Is there a strong requirement that workloads whose certis signed by CertSigner 1 should not talk to workloads whose trustAnchor is signed by CertSigner 2?

Reply:
Make sense, Let me remove this condition check.

Reply:
removed